ironfish/src/rpc/adapters/httpAdapter.test.ts

```diff
@@ -0,0 +1,117 @@
+      enableRpc: false,
+      enableRpcIpc: false,
+      enableRpcTcp: false,
+      enableRpcTls: false,
+      rpcHttpPort: 0,
+    },
+  })
+
+  beforeEach(async () => {
+    httpAdapter = new RpcHttpAdapter('localhost', 0, undefined, ALL_API_NAMESPACES)
+
+    mitm = Mitm()
+    mitm.on('request', (req, res) => httpAdapter?.onRequest(req, res))
+
+    await nodeTest.node.rpc.mount(httpAdapter)
+  }, 20000)
+
+  afterEach(() => {
+    mitm.disable()
+  })
+
+  it('should send and receive message', async () => {
+    Assert.isNotUndefined(httpAdapter)
+    Assert.isNotNull(httpAdapter.router)
+
+    httpAdapter.router.routes.register('foo/bar', yup.string(), (request) => {
+      request.end(request.data)
+    })
+
+    const client = new RpcHttpClient('http://localhost')
+
+    const response = await client.request('foo/bar', 'hello world').waitForEnd()
+    expect(response.content).toBe('hello world')
+  }, 20000)
+
+  it('should stream message', async () => {
+    Assert.isNotUndefined(httpAdapter)
+    Assert.isNotNull(httpAdapter?.router)
+
+    httpAdapter.router.routes.register('foo/bar', yup.object({}), (request) => {
+      request.stream('hello 1')
+      request.stream('hello 2')
+      request.end()
+    })
+
+    const client = new RpcHttpClient('http://localhost')
+
+    const response = client.request('foo/bar')
+    expect((await response.contentStream().next()).value).toBe('hello 1')
+    expect((await response.contentStream().next()).value).toBe('hello 2')
+
+    await response.waitForEnd()
+    expect(response.content).toBe(undefined)
+  }, 20000)
+
+  it('should handle errors', async () => {
+    Assert.isNotUndefined(httpAdapter)
+    Assert.isNotNull(httpAdapter?.router)
+
+    httpAdapter.router.routes.register('foo/bar', yup.object({}), () => {
+      throw new RpcValidationError('hello error', 402, 'hello-error' as ERROR_CODES)
+    })
+
+    const client = new RpcHttpClient('http://localhost')
+
+    const response = client.request('foo/bar')
+
+    await expect(response.waitForEnd()).rejects.toThrow(RpcRequestError)
+    await expect(response.waitForEnd()).rejects.toMatchObject({
+      status: 402,
+      code: 'hello-error',
+      codeMessage: 'hello error',
+    })
+  }, 20000)
+
+  it('should handle request errors', async () => {
+    Assert.isNotUndefined(httpAdapter)
+    Assert.isNotNull(httpAdapter?.router)
+
+    // Requires this
+    const schema = yup.string().defined()
+    // But send this instead
+    const body = undefined
+
+    httpAdapter.router.routes.register('foo/bar', schema, (res) => res.end())
+
+    const client = new RpcHttpClient('http://localhost')
+
+    const response = client.request('foo/bar', body)
+
+    await expect(response.waitForEnd()).rejects.toThrow(RpcRequestError)
+    await expect(response.waitForEnd()).rejects.toMatchObject({
+      status: 400,
+      code: ERROR_CODES.VALIDATION,
+      codeMessage: expect.stringContaining('this must be defined'),
+    })
```

Comment:
Side quest: 
Add test case for timeout.
Add test case for middle of stream failure.

Reply:
I have a good idea. This file was copy pasted from TlsAdapter tests which also don't have those tests. However, all the RPC clients inherit from `RpcClient` base class. We could just make one set of tests that test every adapter and client using a normalized test suite. I can do that in another PR afterwards.

If I write those tests here, they'll only cover the HTTP adapter.

Reply:
SGTM!